pylutron/__init__.py

```diff
@@ -340,39 +378,70 @@ def _parse_keypad(self, keypad_xml, device_group):
       if comp_type == 'BUTTON':
         button = self._parse_button(keypad, comp)
         keypad.add_button(button)
+      elif comp_type == 'CCI':
+        button = self._parse_cci(keypad, comp)
+        keypad.add_button(button)
       elif comp_type == 'LED':
         led = self._parse_led(keypad, comp)
         keypad.add_led(led)
     return keypad
 
   def _parse_button(self, keypad, component_xml):
     """Parses a button device that part of a keypad."""
+    component_number = int(component_xml.get('ComponentNumber'))
     button_xml = component_xml.find('Button')
-    name = button_xml.get('Engraving')
+    engraving = button_xml.get('Engraving')
+    # name = button_xml.get('Engraving')
     button_type = button_xml.get('ButtonType')
     direction = button_xml.get('Direction')
+    led_logic = 0 if button_xml.get('LedLogic') is None else int(button_xml.get('LedLogic'))
+    name = f"Btn {component_number}"
```

Comment:
This is going to significantly change behavior for users of this library.  Why not keep the name as the engraving?

Reply:
1. This is the XML for the CSD.
<img width="912" alt="Screenshot 2024-05-12 alle 10 19 48" src="https://github.com/thecynic/pylutron/assets/29941924/9b87018e-0590-445b-b03e-92e1065bdad2">

As you can see the name is always CSD 001, and there can be multiple keypads in the same area. So CSD 001 is not the name. We could infer the name from the DeviceGroup instead, which is actually the location of the device, as in my XML a deviceGroup contains always only one keypad, but is that true for everyone?
<img width="363" alt="Screenshot 2024-05-12 alle 10 32 32" src="https://github.com/thecynic/pylutron/assets/29941924/ee14d735-6600-43fc-b1cf-8a8127a3ce1c">

2. Phantom keypads. Is not an odd variant, it's included in the documentation and it's a very useful way to create custom button to launch scenes. I agree with you that inferring that it's a phantom keypad from an else is not the best method.

3.  I understand this, but the was an original mistake. The name of the button in my XML is always "Button x", and you cannot customise it. So that should be the name, not the engraving. Using the engraving can be an option on the HA side, as I'm proposing to do with the location name in the area name, and to use the device name in the keypad components and events. Moreover it's not unusual to have multiple keypads on large rooms, that have button with the same engraving, so the name it's not unique.

<img width="1445" alt="Screenshot 2024-05-12 alle 10 33 33" src="https://github.com/thecynic/pylutron/assets/29941924/3fd74a20-b7df-4b77-a6cb-e0b3b441ac56">

Reply:
If we could try to keep the replies attached to the code they're about, it'll make the review easier to manage.

Both RadioRA2 and Homeworks appear to have the same quirk:

1) The button names are practically worthless (non-descriptive)
2) Engravings can be rather common

This nuance is already handled in Home Assistant today.  So I'm trying to understand why you'd want significantly different behavior for Homeworks.  I agree that this library probably should have stayed true to the data model and exported engraving and name as separate properties, and I'd be happy to address that in a separate PR.  But for this one, I think we should stay consistent.

Or to put it another way:  The addition of Homeworks should in no way change the behavior for RadioRA2 users.

Reply:
Why we don't address this problem on the HA side instead? Use a flag "Use Engraving as Button Name", default True, and let people decide? 
In my case I have 34 keypads. Every keypad has btn1 as the main light for the room, with "Luce" as engraving. As it it's I would end up with 34 buttons and scenes with "Luce_x" as entity_id, and I would have to change each entity id to something meaningful.

Reply:
> If we could try to keep the replies attached to the code they're about, it'll make the review easier to manage.

I would, but in github for a lot of comments I cannot add replies, but only at the end of a group.